deb/openmediavault-bcache/usr/share/php/openmediavault/system/storage/bcache/backingdevice.inc

```diff
@@ -0,0 +1,255 @@
+<?php
+/**
+ * This file is part of OpenMediaVault.
+ *
+ * @license   http://www.gnu.org/licenses/gpl.html GPL Version 3
+ * @author    Michael Myrcik <michael.myrcik@web.de>
+ * @copyright Copyright (c) 2009-2024 Volker Theile
+ *
+ * OpenMediaVault is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * OpenMediaVault is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with OpenMediaVault. If not, see <http://www.gnu.org/licenses/>.
+ */
+namespace OMV\System\Storage\Bcache;
+
+/**
+ * Class to handle a bcache backing device.
+ */
+class BackingDevice extends BcacheDevice {
+
+	/**
+	 * Creates a backing device.
+	 * @return void
+	 * @throw \OMV\ExecException
+	 */
+	public function create() {
+		$cmdArgs = [];
+		$cmdArgs[] = "-B";
+		$cmdArgs[] = $this->getDeviceFile();
+		$cmd = new \OMV\System\Process("make-bcache", $cmdArgs);
+		$cmd->setRedirect2to1();
+		$cmd->execute();
+		while(!$this->isRegistered()) {
```

Comment:
IMO there should be an abort condition here.

Reply:
Do you have a suggestion, I would otherwise implement a counter up to 10 with abort?

Reply:
To clarify: The while loop was necessary for a better user experience, as it took some time after creation for the kernel to register the new device, and otherwise the user would see an empty table after creation.

Reply:
Any suggestion on how long the timeout should be specified. I'll use 10 seconds by default if you do not suggest anything else.